### PR TITLE
Add a rake task for diffing POT files

### PR DESCRIPTION
**Pull request by Magisus**
This commit adds a new rake task which will cache the existing POT file
of a project, generate a new one, and check for changes in actual string
content, only replacing the file if relevant changes are found. This
allows us to only update the POT file if necessary and avoid churn due
to changing modification dates.

### CODE REVIEW

**Comment by steveax** [comment]
👍

**Comment by Magisus** [comment]
@smcclellan can you take a look at this? Once it's merged I'll need another release of the gem, since we're starting to pull it into CI jobs that need to use this new task.

File: lib/tasks/gettext.rake [hunk 1]
```diff
@@ -52,10 +51,35 @@ namespace :gettext do
            "--package-version '#{version}' " +
            "--copyright-holder='#{copyright_holder}' --copyright-year=#{Time.now.year} " +
            "#{files_to_translate.join(" ")}")
-    puts "POT file locales/#{project_name}.pot has been updated"
   end
 
-  desc "Update po file for a specific language"
+  desc "Geneate a new POT file and replace old if strings changed"
```
Comment (smcclellan) [inline lib/tasks/gettext.rake]: s/Geneate/Generate/

File: lib/tasks/gettext.rake [hunk 2]
```diff
@@ -52,10 +51,35 @@ namespace :gettext do
...
+  task :update_pot do
+    if !File.exists? pot_file_path
+      puts "No existing POT file, generating new"
+      generate_new_pot
+      next
```
Comment (smcclellan) [inline lib/tasks/gettext.rake]: I think this would be better as an `else` block.

File: lib/tasks/gettext.rake [hunk 3]
```diff
@@ -52,10 +51,35 @@ namespace :gettext do
...
+
+    old_pot = pot_file_path + ".old"
+    File.rename(pot_file_path, old_pot)
+    generate_new_pot
+    if system("msgcmp --use-untranslated '#{old_pot}' '#{pot_file_path}'")
+      puts "No string changes, keeping old POT file"
+      File.rename(old_pot, pot_file_path)
+    elsif
+      File.delete(old_pot)
+      puts "String changes, replacing with updated POT file"
```
Comment (smcclellan) [inline lib/tasks/gettext.rake]: "String changes detected, ..." would be clearer here.

File: lib/tasks/gettext.rake [hunk 4]
```diff
@@ -52,10 +51,35 @@ namespace :gettext do
...
+      puts "No existing POT file, generating new"
+      generate_new_pot
+      next
+    end
+
+    old_pot = pot_file_path + ".old"
+    File.rename(pot_file_path, old_pot)
+    generate_new_pot
+    if system("msgcmp --use-untranslated '#{old_pot}' '#{pot_file_path}'")
+      puts "No string changes, keeping old POT file"
```
Comment (smcclellan) [inline lib/tasks/gettext.rake]: Probably add "...detected, ..." here too.

File: lib/tasks/gettext.rake [hunk 5]
```diff
@@ -52,10 +51,35 @@ namespace :gettext do
...
+    elsif
+      File.delete(old_pot)
+      puts "String changes, replacing with updated POT file"
+    end
+  end
+
+  desc "Generate POT file"
+  task :pot do
+    generate_new_pot
+    puts "POT file locales/#{GettextSetup.config['project_name']}.pot has been generated"
```
Comment (smcclellan) [inline lib/tasks/gettext.rake]: Why not just use `#{pot_file_path}` in this string?

**Comment by Magisus** [comment]
@smcclellan updated!

**Pull request merged**